src/DependencyInjection/Configuration.php

```diff
@@ -166,6 +171,31 @@ function (string $value): string {
         return $treeBuilder;
     }
 
+    /**
+     * Returns the available Imagine implementation.
+     *
+     * @return string
+     */
+    private function getImagineImplementation(): string
+    {
+        static $magicks = ['Gmagick', 'Imagick'];
+
+        foreach ($magicks as $name) {
+            $class = 'Imagine\\'.$name.'\Imagine';
+
+            // Will throw an exception if the PHP implementation is not available
+            try {
+                new $class();
+            } catch (RuntimeException $e) {
```

Comment:
What's the benefit of catching instead of letting it fail?

Reply:
It will fall back to `return Imagine::class; // see #616`